c10/core/Layout.h

```diff
@@ -14,7 +14,8 @@ enum class Layout : int8_t {
   SparseCsc,
   SparseBsr,
   SparseBsc,
-  NumOptions
+  NumOptions,
+  Unspecified = NumOptions
```

Comment:
I don't think we should add this just so we can write maybe slightly better code within sparse. It seems too global, generic and intrusive for a local problem.

Reply:
I have moved `Unspecified` under the `at::sparse_csr` namespace as `kUnspecified`.

Reply:
`kUnspecified` is now eliminated.